riscv-c-api.md

```diff
@@ -39,6 +39,71 @@ https://creativecommons.org/licenses/by/4.0/.
 | __riscv_32e         | 1     | `E` extension is available.   |
+| __riscv_v_min_vlen    | <N>   | `V` or `Zve*` extensions are available, more detailed description see [__riscv_v_min_vlen](#__riscv_v_min_vlen) |
+| __riscv_v_max_eew     | <N>   | `V` or `Zve*` extensions are available, more detailed description see [__riscv_v_max_eew](#__riscv_v_max_eew) |
+| __riscv_v_max_eew_fp  | <N>   | `V` or `Zve*` extensions are available, more detailed description see [__riscv_v_max_eew_fp](#__riscv_v_max_eew_fp) |
+
+
+### __riscv_v_min_vlen
+
+This macro present the minimal VLEN in bit, which is guaranteed by the
+architecture extension, e.g. `V` extension requires that VLEN ≥ 128, `Zve32*`
+extension requires VLEN ≥ 32 and `Zve64*` extension requires VLEN ≥ 64; also
+RISC-V have additional `Zvl*b` extensions to guarantee the minimal VLEN, that
+will effect the value of `__riscv_v_min_vlen` too.
+
+So the value of `__riscv_v_min_vlen` is defined by following rules:
+
+- 128 if `V` extension is available.
+- 32 if `Zve32x` or `Zve32f` extension is available.
+- 64 if `Zve64x`, `Zve64f` or `Zve64d` extension is available.
+- `N` if `Zvl<N>b` extension is available, and if `N` is large than the value
+  defined above.
+- `__riscv_v_min_vlen` is undefined if all of above rule are not match.
+
+Example:
+
+- `__riscv_v_min_vlen` is 128 for `rv64gcv`
+- `__riscv_v_min_vlen` is 512 for `rv32gcv_zv512b`
+- `__riscv_v_min_vlen` is 256 for `rv32gcv_zvl32b_zv256b`
+- `__riscv_v_min_vlen` is 128 for `rv64gcv_zvl32b`
+
+### __riscv_v_max_eew
+
+This marco present the maximal EEW (effective element width) value for
+non-floating-point operations, which is guaranteed by the architecture
+extension, e.g. `V` and `Zve64*` extension require EEW of 8, 16, and 32, and 64,
+but `Zve32*` only require EEW of 8, 16, and 32.
+
+The value of `__riscv_v_max_eew`:
+
+- `__riscv_v_max_eew` is 64 if `V`, `Zve64x`, `Zve64f` or `Zve64d` extension
+  is available.
+- `__riscv_v_max_eew` is 32 if `Zve32x` or `Zve32f` extension is available.
+- `__riscv_v_max_eew` is undefined if all of above rule are not match.
+
+NOTE: Toolchain may support integer vector type large than `__riscv_v_max_eew`,
+      but it migth cause illegal instruction exception at run-time.
+
+### __riscv_v_max_eew_fp
+
+This marco present the maximal EEW value for floating-point operations,
+which is guaranteed by the architecture extension, e.g. Support of
+floating-point operands with EEW=32 or EEW=64 are required by `V` extension and
+`zve64d` extension, `Zve32f` and `Zve64f` only require support
+floating-point operations with EEW=32, and `Zve32x` and `Zve64x` are not supprot
+floating-point operations.
+
+The value of `__riscv_v_max_eew_fp`:
+
+- `__riscv_v_max_eew_fp` is 64 if `V` or `Zve64d` extension is available.
+- `__riscv_v_max_eew_fp` is 32 if `Zve32f` or `Zve64f` extension is available.
+- `__riscv_v_max_eew_fp` is 0 if `Zve64x` or `Zve32x` extension is available.
```

Comment:
Why does it define as 0, instead of undefined?

Reply:
My intention is make sure this marco is always defined if vector extension is available, so we only need to check `__riscv_v_max_eew_fp` is larger than our requirement once we ensure the vector extension is available.

e.g.
```
#ifdef __riscv_v_max_eew
#error "NO vector extension supported"
#endif

#if __riscv_v_max_eew_fp > 64
...
#endif
```

Rather than
```
#ifdef __riscv_v_max_eew
#error "NO vector extension supported"
#endif

#if defined(__riscv_v_max_eew_fp) && __riscv_v_max_eew_fp >= 64
...
#endif
```